src/Http/Routing/src/DefaultLinkGenerator.cs

```diff
@@ -336,137 +336,92 @@
-                LogLevel.Debug,
-                EventIds.TemplateFailedRequiredValues,
-                "Failed to process the template {Template} for {Endpoint}. " +
-                "A required route value is missing, or has a different value from the required default values. " +
-                "Supplied ambient values {AmbientValues} and {Values} with default values {Defaults}",
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, string, string, IRouteConstraint, string, string, Exception> _templateFailedConstraint = LoggerMessage.Define<string, string, IRouteConstraint, string, string>(
-                LogLevel.Debug,
-                EventIds.TemplateFailedConstraint,
-                "Failed to process the template {Template} for {Endpoint}. " +
-                "The constraint {Constraint} for parameter {ParameterName} failed with values {Values}",
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, string, string, string, Exception> _templateFailedExpansion = LoggerMessage.Define<string, string, string>(
-                LogLevel.Debug,
-                EventIds.TemplateFailedExpansion,
-                "Failed to process the template {Template} for {Endpoint}. " +
-                "The failure occurred while expanding the template with values {Values} " +
-                "This is usually due to a missing or empty value in a complex segment",
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, IEnumerable<string>, string, Exception> _linkGenerationSucceeded = LoggerMessage.Define<IEnumerable<string>, string>(
-                LogLevel.Debug,
-                EventIds.LinkGenerationSucceeded,
-                "Link generation succeeded for endpoints {Endpoints} with result {URI}",
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, IEnumerable<string>, Exception> _linkGenerationFailed = LoggerMessage.Define<IEnumerable<string>>(
-                LogLevel.Debug,
-                EventIds.LinkGenerationFailed,
-                "Link generation failed for endpoints {Endpoints}",
-                skipEnabledCheck: true);
-
-            public static void EndpointsFound(ILogger logger, object address, IEnumerable<Endpoint> endpoints)
-            {
-                // Checking level again to avoid allocation on the common path
-                if (logger.IsEnabled(LogLevel.Debug))
-                {
-                    _endpointsFound(logger, endpoints.Select(e => e.DisplayName), address, null);
-                }
-            }
+            [LoggerMessage(EventId = 100, EventName = "EndpointsFound", Level = LogLevel.Debug, Message = "Found the endpoints {Endpoints} for address {Address}")]
+            public static partial void EndpointsFound(ILogger logger, object address, IEnumerable<Endpoint> endpoints);
```

Comment:
Does the order of `{Endpoints}` and `{Address}` need to match in the string with how it is ordered in the method signature?

If not, do the names need to match? Here they differ by case.

Reply:
I think the generator looks them up by name (case insensitively).

Reply:
Need to check.

Reply:
Here's the dope code it generates: https://gist.github.com/pranavkm/dba57e845227045d8e3651b0d49d3418